plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/notifications/NotificationFormatUtilsTest.kt

```diff
@@ -0,0 +1,136 @@
+
+    @BeforeAll
+    fun setUp() {
+        mockSystemDetails = SystemDetails(
+            computeType = "Local",
+            computeArchitecture = "x86_64",
+            osType = "Linux",
+            osVersion = "5.4.0",
+            ideType = "IC",
+            ideVersion = "2023.1",
+            pluginVersions = mapOf(
+                "aws.toolkit" to "1.0",
+                "amazon.q" to "2.0"
+            )
+        )
+
+        exampleNotification = javaClass.getResource("/exampleNotification2.json")?.let {
+            Paths.get(it.toURI()).takeIf { f -> f.exists() }
+        }?.inputStream() ?: throw RuntimeException("Test not found")
+
+        mockkStatic("software.aws.toolkits.jetbrains.core.notifications.RulesEngineKt")
+        every { getCurrentSystemAndConnectionDetails() } returns mockSystemDetails
+        every { getConnectionDetailsForFeature(projectRule.project, BearerTokenFeatureSet.Q) } returns FeatureAuthDetails(
+            "Idc",
+            "us-west-2",
+            "Connected"
+        )
+    }
+
+    @AfterAll
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `test System Details`() {
+        val result = getCurrentSystemAndConnectionDetails()
+        assertThat(mockSystemDetails).isEqualTo(result)
+    }
+
+    @Test
+    fun `check Json Validity which has all the required fields`() {
+        assertDoesNotThrow {
+            mapper.readValue<NotificationsList>(exampleNotification)
+        }
+    }
+
+    @Test
+    fun `No schema version associated with the notification file throws an exception`() {
+        assertThrows<Exception> {
+            mapper.readValue<NotificationsList>(exampleNotificationWithoutSchema)
+        }
+    }
+
+    @Test
+    fun `No notifications present with the version file does not throw an exception`() {
+        assertDoesNotThrow {
+            mapper.readValue<NotificationsList>(exampleNotificationWithoutNotification)
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("validNotifications")
+    fun `The notification is shown`(notification: String) {
+        val notificationData = mapper.readValue<NotificationData>(notification)
+        val shouldShow = RulesEngine.displayNotification(notificationData, projectRule.project)
+        assertThat(shouldShow).isTrue
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidNotifications")
+    fun `The notification is not shown`(notification: String) {
+        val notificationData = mapper.readValue<NotificationData>(notification)
+        val shouldShow = RulesEngine.displayNotification(notificationData, projectRule.project)
+        assertThat(shouldShow).isFalse
+    }
+
+    companion object {
+        @JvmStatic
+        fun validNotifications(): Stream<Arguments> = Stream.of(
+            Arguments.of(notificationWithConditionsOrActions),
+            Arguments.of(notificationWithoutConditionsOrActions),
+            Arguments.of(notificationWithValidConnection)
```

Comment:
you should also assert that the deserialized object matches what you're expecting